### PR TITLE
Use the newly updated GPG key for GitHub CLI

### PR DESCRIPTION
**Pull request by exilvm**
Fixes the [`apt-key` approach](https://github.com/devcontainers/features/blob/a5fbdab3c096152dd3a318c0c23da9465815aa69/src/github-cli/install.sh#L230) (related to #132) by using the new GPG key: https://github.com/cli/cli/issues/6175#issuecomment-1238477714.

### CODE REVIEW

**Review by jkeech — DISMISSED** [review]
Thanks! Do you mind also bumping the patch version in the `devcontainer-feature.json`?

File: src/github-cli/install.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@
 CLI_VERSION=${VERSION:-"latest"}
 INSTALL_DIRECTLY_FROM_GITHUB_RELEASE=${INSTALLDIRECTLYFROMGITHUBRELEASE:-"true"}
 
-GITHUB_CLI_ARCHIVE_GPG_KEY=C99B11DEB97541F0
+GITHUB_CLI_ARCHIVE_GPG_KEY=23F3D4EA75716059
```
Comment (samruddhikhandale) [inline src/github-cli/install.sh:13]: Can we bump the patch version here to 1.0.4? We follow semantic versioning. https://github.com/devcontainers/features/blob/main/src/github-cli/devcontainer-feature.json#L3

Features are released every time a PR is merged, these changes won't take effect unless the version specified here is different than the ones already published 😄

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thanks for making this change ✨

Requested one change, besides that LGTM.

File: src/github-cli/install.sh (hunk 1 above)
Comment (exilvm) [reply src/github-cli/install.sh:13]: Thanks for letting me know!

**Review by jkeech — APPROVED** [review]

**Review by samruddhikhandale — APPROVED** [review]
Thanks! 👏

**Pull request merged**